content/docs/command-reference/fetch.md

```diff
@@ -42,20 +42,22 @@ project's cache                  ++ | dvc pull |
  workspace
 ```
 
-Fetching could be useful when first checking out a <abbr>DVC project</abbr>,
-since files tracked by DVC should already exist in remote storage, but won't be
-in the project's <abbr>cache</abbr>. (Refer to `dvc remote` for more information
-on DVC remotes.) These necessary data or model files are listed as
-<abbr>dependencies</abbr> or <abbr>outputs</abbr> in a target
-[stage](/doc/command-reference/run) (in `dvc.yaml`) or `.dvc` file, so they are
-required to [reproduce](/doc/tutorials/get-started/data-pipelines#reproduce) the
-corresponding [pipeline](/doc/command-reference/pipeline).
-
-`dvc fetch` ensures that the files needed for a stage or `.dvc` file to be
-[reproduced](/doc/tutorials/get-started/data-pipelines#reproduce) exist in
-cache. If no `targets` are specified, the set of data files to fetch is
-determined by analyzing all `dvc.yaml` and `.dvc` files in the current branch,
-unless `--all-branches` or `--all-tags` is specified.
+Fetching is useful when first checking out a <abbr>DVC project</abbr> for
+example, to get any files tracked by DVC that already exist in remote storage
+(see `dvc push`) to the local <abbr>cache</abbr>. Refer to `dvc remote` for more
+information on DVC remotes.
+
+`dvc fetch` ensures that the files needed for all the stages (in
+[`dvc.yaml`](/doc/user-guide/dvc-files-and-directories#dvcyaml-file)) and
+[`.dvc` files](/doc/user-guide/dvc-files-and-directories#dvc-files) exist in the
+cache. It can be limited to any given `targets` (files inside directories
+[tracked as a whole](/doc/command-reference/add#example-directory) are
+supported). These data files, datasets, or models are listed as
+<abbr>outputs</abbr>.
+
+If no `targets` are specified, the set of data files to fetch is determined by
+scanning all `dvc.yaml` and `.dvc` files in the workspace (the `--all-branches`
```

Comment:
dvc.lock?

Reply:
Yes, makes more sense to me as well although I'm not sure which one is correct technically. I asked Ruslan about this in https://github.com/iterative/dvc.org/pull/1384#pullrequestreview-451265028 — resolving here.